packages/headless/src/features/commerce/recommendations/recommendations-actions.ts

```diff
@@ -0,0 +1,66 @@
+import {createAction, createAsyncThunk} from '@reduxjs/toolkit';
+import {AsyncThunkCommerceOptions} from '../../../api/commerce/commerce-api-client';
+import {CommerceRecommendationsRequest} from '../../../api/commerce/recommendations/recommendations-request';
+import {RecommendationsCommerceSuccessResponse} from '../../../api/commerce/recommendations/recommendations-response';
+import {isErrorResponse} from '../../../api/search/search-api-client';
+import {validatePayload} from '../../../utils/validate-payload';
+import {logQueryError} from '../../search/search-analytics-actions';
+import {
+  StateNeededByQueryRecommendationsCommerceAPI,
+  buildCommerceAPIRequest,
+} from '../common/actions';
+import {recommendationsSlotDefinition} from './recommendations';
+
+export interface QueryRecommendationsCommerceAPIThunkReturn {
+  /** The successful recommendations response. */
+  response: RecommendationsCommerceSuccessResponse;
+}
+
+const buildRecommendationCommerceAPIRequest = async (
+  slotId: string,
+  state: StateNeededByQueryRecommendationsCommerceAPI
+): Promise<CommerceRecommendationsRequest> => {
+  const commerceAPIRequest = await buildCommerceAPIRequest(state);
+  delete commerceAPIRequest.facets;
```

Comment:
Do we still need this?
```suggestion
```

Reply:
Facets are always included in the type, so changing this introduces a bigger change to the `CommerceAPIRequest` object

Reply:
Maybe since we have a `BaseCommerceAPIRequest`, we can create a `buildBaseCommerceAPIRequest` and use that inside `buildCommerceAPIRequest` and `buildRecommendationCommerceAPIRequest` to build the shared properties then?

Reply:
Yep, I think I have that working now